lib/index.js

```diff
@@ -469,13 +500,39 @@ NodeHttpXHR.prototype.send = function (data) {
     this._resp = resp;
     this._responseText = '';
 
-    resp.setEncoding('utf8');
+    // var contentType = resp.headers['content-type'];
+    // TODO: adjust responseType from content-type header if applicable
+
+    // from Node API docs: The encoding argument is optional and only applies when chunk is a string. Defaults to 'utf8'.
+    if (this._responseType === 'text' || this._responseType === '') {
+      resp.setEncoding('utf8');
+    }
+
     resp.on('data', function onData(chunk) {
-      this._responseText += chunk;
+
+      if (typeof chunk === 'string') {
+        this._responseText += chunk;
+      } else if (typeof chunk === 'object') {
+        if (chunk instanceof Buffer) {
+          if (this._response) {
+            this._response = Buffer.concat([this._response, chunk]);
+          } else {
+            this._response = chunk;
+          }
+        // binary data usually comes in one chunk (no chunky transfer-encoding)
```

Comment:
note to self: this comment is confusing. transfer-encoding is unrelated to the chunks of data delivered by this callback (which is reading from net socket)